src/boot/efi/stub.c

```diff
@@ -180,6 +181,139 @@ static bool use_load_options(
         return true;
 }
 
+static EFI_STATUS get_cmdline_addon(
+                EFI_HANDLE stub_image,
+                EFI_LOADED_IMAGE_PROTOCOL *loaded_image,
+                char16_t **ret_buffer) {
+
+        _cleanup_(file_closep) EFI_FILE *root = NULL, *extra_dir = NULL;
+        size_t dirent_size = 0, n_items = 0, n_allocated = 0;
+        _cleanup_free_ EFI_FILE_INFO *dirent = NULL, *extra_dir_info = NULL;
+        _cleanup_(strv_freep) char16_t **items = NULL;
+        _cleanup_free_ char16_t *buffer = NULL;
+        EFI_STATUS err;
+
+        assert(stub_image);
+        assert(loaded_image);
+        assert(ret_buffer);
+
+        if (!loaded_image->DeviceHandle)
+                goto nothing;
+
+        err = open_volume(loaded_image->DeviceHandle, &root);
+        if (err == EFI_UNSUPPORTED)
+                /* Error will be unsupported if the bootloader doesn't implement the file system protocol on
+                 * its file handles. */
+                goto nothing;
+        if (err != EFI_SUCCESS)
+                return log_error_status(err, "Unable to open root directory: %m");
+
+        err = open_directory(root, u"\\loader\\addons", &extra_dir);
```

Comment:
Please add logic right away for per-kernel-image cmdline files. I am pretty sure we should advertise the per-kernel studd before the gloabl settings really

Reply:
I wanted to keep that for later, but ok, I'll try to find the time to add this too

Reply:
ok, added